main.cpp

```diff
@@ -18,6 +18,8 @@ struct TCB {
     sigjmp_buf env;
     thread_func function;
     void* arg;
+    void* result;
+    int waiting_for_tid = -1;
     bool complete;
```

Comment:
If complete=true means that the thread has finished executing, why do you set it to true in the thread_wrapper?

Reply:
@kstowe Is there something wrong with setting it to complete there?
There is the issue of race conditions,, but we can turn off preemption during that section of code, or do something more complex.

Reply:
We set complete to be true in the wrapper, because that is the last statement that executes before the thread returns.

Reply:
No you're right. Somehow I didn't see the function call.